### PR TITLE
baurmc: add new package

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/baurmc/package.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    def do_stage(self, mirror_only=False):
+        # the tarball extracts to an intermediate directory -
+        # move everything to the proper source dir
+        super(Baurmc, self).do_stage(mirror_only)
+        dn = os.listdir(self.stage.source_path)[0]
+        for fn in os.listdir(join_path(self.stage.source_path, dn)):
+            shutil.move(join_path(self.stage.source_path, dn, fn),
+                        join_path(self.stage.source_path, fn))
+        shutil.rmtree(join_path(self.stage.source_path, dn))
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/baurmc/package.py]: Can't you set the `configure_path` property instead of moving everything?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/baurmc/package.py (hunk 1 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/baurmc/package.py]: Do you have a link to docs or an example? (Grepping the builtin spack repo for `configure_path` comes up empty)

Comment (alalazo) [reply var/spack/repos/builtin/packages/baurmc/package.py]: Apologies, I meant `configure_directory`:
https://github.com/spack/spack/blob/62abbeaf6f9e6aaa8f34baa2ef1ee57139577ea8/lib/spack/spack/build_systems/autotools.py#L285-L291

Comment (vvolkl) [reply var/spack/repos/builtin/packages/baurmc/package.py]: Ah, I see , thanks for the pointer.

However, the intermediate directory is names after the version (extracts to `spack-src/1.0/configure`) so I'm not sure hardcoding that is an improvmement.

Comment (alalazo) [reply var/spack/repos/builtin/packages/baurmc/package.py]: I'm not sure I understand the issue. Wouldn't:
```python
@property 
def configure_directory(self): 
    return os.path.join(self.stage.source_path, str(self.spec.version))
```
be fine?

Comment (vvolkl) [reply var/spack/repos/builtin/packages/baurmc/package.py]: Yes, that works, thanks!

File: var/spack/repos/builtin/packages/baurmc/package.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                    string=True)
+
+    def configure_args(self):
+        options = ['--userfflags=-fno-automatic',
+                   '--enable-shared',
+                   ]
+        return options
+
+    def install(self, a, b):
+        install_tree(str(self.spec.version) + '/lib', self.prefix.lib)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/baurmc/package.py]:
```suggestion
    def install(self, spec, prefix):
        build_libdir = os.path.join(str(spec.version), "lib")
        install_tree(build_libdir, self.prefix.lib)
```

File: var/spack/repos/builtin/packages/baurmc/package.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        filter_file('FC=g77',
+                    'FC=gfortran',
+                    str(self.spec.version)  + '/configure',
+                    string=True)
+
+    def configure_args(self):
+        options = ['--userfflags=-fno-automatic',
+                   '--enable-shared',
+                   ]
+        return options
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/baurmc/package.py]:
```suggestion
        return [
            '--userfflags=-fno-automatic',
            '--enable-shared'
        ]
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Acouple of minor changes, but otherwise LGTM.

**Review by alalazo — APPROVED** [review]

**Pull request merged**